### PR TITLE
Develop Bubble Sort Algorithm

### PR DESCRIPTION
**Pull request by MarilenaRoque**
On this branch, we have create two methods

- bubble_sort: it sorts an array of numbers using the bubble sort technique
-bubble_sort_by: It sorts an array of strings using its lengths 

We have applied the theory learned about yield and blocks for both methods

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+      test = yield(numb, new_array[position + 1])
+      next unless test.positive?
+
+      new_array[position] = new_array[position + 1]
+      new_array[position + 1] = numb
+      flag = 0
+    end
+  end
+end
```
Comment (shshamim63) [inline bubble_sort.rb]: - Sort returns the sorted array. So your bubble_sort_by should do the same task. But it is not returning anything. That's why while i tried to execute the following code it did not work.
```ruby
print bubble_sort_by([1, 5, 3, 2, 1]) {|num1, num2| num1 - num2}
```
So, after line 32 you should place new_array.

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
```
Comment (shshamim63) [inline bubble_sort.rb]: - bubble_sort does not interact with any block. So yield is redundant. Instead, it should compare two items.
 - Sort returns the sorted array. So your bubble_sort should do the same task. But it is not returning anything. That's why while i tried to execute the following code it did not work.
```ruby
print bubble_sort([1, 5, 3, 2, 1])
```
So, after line 15 you should place new_array.

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :1234: Changes Required
___
Hi @MarilenaRoque and @yahir91 :wave: 
Almost completed the tasks. just need to cover some edge cases. Let's get to them.
___
### :spades: Mandatory Chagnes :spades: 
___
- Check the comments below ans resolve the issue
___
##### After resolving all the issues please send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

File: bubble_sort.rb (hunk 2 above)
Comment (yahir91) [reply bubble_sort.rb]: We implemented the next changes

Added new_array to both methods, now it prints both results.
We fixed the redundant yield on bubble_sort method.

Thank you for your review.

**Comment by MarilenaRoque** [comment]
We have fixed all the issues pointed:

- [x] Both methods return the array
- [x] Removed yield for bubble_sort method

Thank you so much for the review @shshamim63!

**Review by serdg0 — APPROVED** [review]
**Congratulations** @MarilenaRoque @yahir91  :fireworks: :100: :fireworks:
Feel free to merge.
Keep the good work in your next project!
**Happy Coding!** :computer: :nerd_face:
**_Sergio Diaz_**

**Comment by MarilenaRoque** [comment]
Thank you so much for the review @serdg0! :rocket:

**Pull request merged**